core/block_svg.ts

```diff
@@ -436,16 +444,31 @@ export class BlockSvg
    * @returns Object with coordinates of the bounding box.
    */
   getBoundingRectangle(): Rect {
+    return this.getBoundingRectangleWithWidth(this.getHeightWidth().width);
+  }
+
+  /**
+   * Returns the coordinates of a bounding box describing the dimensions of this
+   * block alone.
+   * Coordinate system: workspace coordinates.
+   *
+   * @returns Object with coordinates of the bounding box.
+   */
+  getBoundingRectangleWithoutChildren(): Rect {
+    return this.getBoundingRectangleWithWidth(this.childlessWidth);
+  }
+
+  private getBoundingRectangleWithWidth(blockWidth: number) {
     const blockXY = this.getRelativeToSurfaceXY();
     const blockBounds = this.getHeightWidth();
     let left;
     let right;
     if (this.RTL) {
-      left = blockXY.x - blockBounds.width;
+      left = blockXY.x - blockWidth;
       right = blockXY.x;
     } else {
       left = blockXY.x;
-      right = blockXY.x + blockBounds.width;
+      right = blockXY.x + blockWidth;
     }
     return new Rect(blockXY.y, blockXY.y + blockBounds.height, left, right);
```

Comment:
Does this still have the problem with the height changing depending on whether the block is selected or not?

Reply:
Nope, results are consistent with the old `getBoundingRectangle()` method except when the block has children/next blocks.